### ISSUE TITLE
Compile error

### ISSUE THREAD

**Issue by akhalili** [augcog/OpenARK#38]
1- ***************************************************************************************************
I get the following compile error during the build with cmake.
I am using Eigen 3.3.4 as mentioned in the installation guide. Any idea how to solve?

```
OpenARK/include/Util.h: In function ‘Eigen::Matrix<_Scalar, 3, 1> ark::util::rotate(const Eigen::Matrix<_Scalar, 3, 1>&, const Quat_T&)’:
OpenARK/include/Util.h:655:62: error: expected primary-expression before ‘>’ token
             const Eigen::Matrix<T, 3, 1> & u = q.vec().cast<T>();
```

2- ****************************************************************************************************
I am wondering where pcl and Eigen are included in /include/Util.h. My compiler cannot find them

**Comment by akhalili** [comment]
Regarding (1) fix it by changing:

`q.vec().cast<T>();`

to 

`q.vec().template cast<T>();`

**Comment by dahburj** [comment]
I've spent the entire day trying to get this to build. All the packages need to be hacked/fixed in one weird way or another so far, and it STILL won't build. Thank you for the Eigen fix, I'm still getting problems with PCL. I remember seeing one of the requirements listing opencv < 3.0 in it's requirements. I'm now getting "pcl does not name a type"  "Eigen does not name a type" errors. Have you gotten the package to work at all?

**Comment by akhalili** [comment]
That's right. It needs many hacks. I have built the following commit successfully on a linux 
`adcf6db0b5301be0253c1c1cfd897d5ec3534aa3`

I only activate the building of the avatar. It would be my pleasure to share my modified branch.

Regarding the unidentified Eigen and PCL, either add  #include "stdafx.h" on top of the file, or add the appropriate Eigen or PCL header file.

I used opencv 3.4.6 built from source with opencv_contrib

The code still will have many windows based hardcoded paths that I needed to change
